### PR TITLE
JAMES-3604 Improve RabbitMQ resiliency

### PR DESCRIPTION
**Pull request by chibenwa**
- [x] Allow the use of Quorum queues for the EventBus and the MailQueues
 - [x] Use publish confirms as recommended
 - [x] Allow via the configuration to pass several contact points
 - [x] Measure performance costs of publish confirms

To be done later:

 - [ ] Once we deployed a RabbitMQ cluster in one of our environments at Linagora, and once we did deploy Quorum queues with a replication factor of 3, we will perform a quick return of experience.

### CODE REVIEW

**Comment by chibenwa** [comment]
CPU utilisation related to RabbitMQ as expected increased (1.30 % -> 2.6%) and we lost a few ms in latencies.

Given that it is required for correctness, this sounds like a decent compromise.

**Comment by chibenwa** [comment]
Undrafted as I move publish/confirms to a separated PR...

File: server/mailet/integration-testing/src/test/java/org/apache/james/transport/mailets/AmqpForwardAttachmentTest.java [hunk 1]
```diff
@@ -71,6 +71,7 @@
 
     @BeforeEach
     void setup(@TempDir File temporaryFolder) throws Exception {
+        System.out.println(amqpExtension.getAmqpUri());
```
Comment (Arsnael) [inline server/mailet/integration-testing/src/test/java/org/apache/james/transport/mailets/AmqpForwardAttachmentTest.java:74]: debug spotted

**Review by Arsnael — APPROVED** [review]

File: server/apps/distributed-app/sample-configuration/rabbitmq.properties [hunk 2]
```diff
@@ -5,6 +5,10 @@
 # Mandatory
 uri=amqp://rabbitmq:5672
```
Comment (vttranlina) [inline server/apps/distributed-app/sample-configuration/rabbitmq.properties:6]: Should we deprecate `uri` property? 
When we declare `hosts` property, what is `uri` sense?
Should we have only one property for endpoint declaration? IMO

File: src/site/xdoc/server/config-rabbitmq.xml [hunk 3]
```diff
@@ -118,6 +118,11 @@
 
           <dt><strong>quorum.queues.replication.factor</strong></dt>
           <dd> Strictly positive integer. The replication factor to use when creating quorum queues.</dd>
+
+          <dt><strong>hostsr</strong></dt>
```
Comment (vttranlina) [inline src/site/xdoc/server/config-rabbitmq.xml:122]: `hosts`

File: server/apps/distributed-app/sample-configuration/rabbitmq.properties (hunk 2 above)
Comment (chibenwa) [reply server/apps/distributed-app/sample-configuration/rabbitmq.properties:6]: > Should we deprecate uri property?

If I can avoid a breaking change I'm happy.

I think URI is still easier to use for a single Rabbit set up.

You can further more pass extra info in it: VHost, username + password, etc...

> When we declare hosts property, what is uri sense?

 1. Host is optional
 2. URI can be used for vhost, username, password, etc...
  
> Should we have only one property for endpoint declaration? IMO

We have to balance this with other goals too: how about avoiding to much breaking changes?

Comment (vttranlina) [reply server/apps/distributed-app/sample-configuration/rabbitmq.properties:6]: Thanks for answer

**Comment by chibenwa** [comment]
Just squashed

**Review by Arsnael — APPROVED** [review]

**Comment by Arsnael** [comment]
Why closed?

**Comment by chibenwa** [comment]
Some build errors, got that on another PR and I was forced to reopen it to sole it. Looked like a corrupted .m2...

**Comment by Arsnael** [comment]
Yeah I just saw that... What can we do about it, as the corrupted m2 package is on apache jenkins?

**Pull request closed without merging**